src/Raven.Server/Documents/ETL/Providers/RelationalDatabase/Common/RelationalWriters/RelationalDatabaseWriterBase.cs

```diff
@@ -0,0 +1,574 @@
+                    }
+
+                    throw new InvalidOperationException("Cannot understand how to save " + column.Type + " for " + colParam.ParameterName);
+                }
+            }
+        }
+    }
+
+    private static void SetParamStringValue(DbParameter colParam, string value, List<Func<DbParameter, string, bool>> stringParsers)
+    {
+        if (value.Length > 0 && stringParsers != null)
+        {
+            foreach (var parser in stringParsers)
+            {
+                if (parser(colParam, value))
+                {
+                    return;
+                }
+            }
+        }
+
+        colParam.Value = value;
+    }
+
+    public List<Func<DbParameter, string, bool>> GenerateStringParsers()
+    {
+        return new List<Func<DbParameter, string, bool>>
+        {
+            (colParam, value) =>
+            {
+                if (char.IsDigit(value[0]))
+                {
+                    if (DateTime.TryParseExact(value, DefaultFormat.OnlyDateTimeFormat, CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind,
+                            out DateTime dateTime))
+                    {
+                        colParam.Value = dateTime;
+                        return true;
+                    }
+                }
+
+                return false;
+            },
+            (colParam, value) =>
+            {
+                if (char.IsDigit(value[0]))
+                {
+                    if (DateTimeOffset.TryParseExact(value, DefaultFormat.DateTimeFormatsToRead, CultureInfo.InvariantCulture,
+                            DateTimeStyles.RoundtripKind, out DateTimeOffset dateTimeOffset))
+                    {
+                        colParam.Value = dateTimeOffset;
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        };
+    }
+
+    public DbCommand GetInsertCommand(string tableName, string pkName, RelationalDatabaseItem itemToReplicate)
+    {
+        var cmd = CreateCommand();
+        
+        var sb = new StringBuilder("INSERT INTO ")
+            .Append(GetTableNameString(tableName))
+            .Append(" (")
+            .Append(_commandBuilder.QuoteIdentifier(pkName))
+            .Append(", ");
+        foreach (var column in itemToReplicate.Columns)
+        {
+            if (column.Id == pkName)
+                continue;
+            sb.Append(_commandBuilder.QuoteIdentifier(column.Id)).Append(", ");
+        }
+
+        sb.Length = sb.Length - 2;
+
+        var pkParam = cmd.CreateParameter();
+        pkParam.ParameterName = GetParameterNameForDbParameter(pkName);
+
+        SetPrimaryKeyParamValue(itemToReplicate, pkParam);
+        cmd.Parameters.Add(pkParam);
+
+        var syntax = GetSyntaxAroundParameters(itemToReplicate);
+
+        sb.Append($") {syntax.StartSyntax}");
+
+        sb.Append(GetParameterNameForCommandString(pkName, false)).Append(", ");
```

Comment:
can you add name to `false` argument ? Other places as well.

Reply:
done